editor/editor_settings.cpp

```diff
@@ -873,6 +873,22 @@ bool EditorSettings::_is_default_text_editor_theme(String p_theme_name) {
 	return p_theme_name == "default" || p_theme_name == "godot 2" || p_theme_name == "custom";
 }
 
+Ref<ConfigFile> EditorSettings::_load_project_metadata() const {
+	if (project_metadata.is_null()) {
+		project_metadata.instantiate();
+	}
+
+	String path = EditorPaths::get_singleton()->get_project_settings_dir().path_join("project_metadata.cfg");
+	// Load project metadata if not already loaded or if modified externally.
+	if (int(project_metadata->get_meta(SNAME("last_modified_time"), 0)) < FileAccess::get_modified_time(path)) {
```

Comment:
To avoid some potential issues when people use USB-sticks or network drives to store their projects, I would maybe use `!=` comparison instead of `<` here to compare timestamps. Some filesystems have bugs or too coarse timestamps precision that might get rounded strangely. We only really care if the modfied timestamp has changed, even if it went backwards it probably means the file was modified somehow.

I would also avoid calling `FileAccess::get_modified_time()` twice, as some drives might be pretty slow or don't cache this information. So maybe something like this? (untested):

```cpp
String path = EditorPaths::get_singleton()->get_project_settings_dir().path_join("project_metadata.cfg");
// Load project metadata if not already loaded or if modified externally.
uint64_t modified_time = FileAccess::get_modified_time(path);
if (uint64_t(project_metadata->get_meta(SNAME("last_modified_time"), 0)) != modified_time) {
    Error err;
    err = project_metadata->load(path);
    ERR_FAIL_COND_V_MSG(err != OK && err != ERR_FILE_NOT_FOUND, project_metadata, "Cannot load project metadata from file '" + path + "'.");
    project_metadata->set_meta(SNAME("last_modified_time"), modified_time);
}
```

Reply:
If the timestamp is wrong for whatever reason, the file will be simply loaded again.
Also I wouldn't worry about checking modified time if previously the editor would load and parse the whole file.

Reply:
Btw, sorry if this feels like random drive-by review, I'm sometimes just reading interesting new PR's and now I'm wondering if I missed some aspect of the this loading code myself.

>If the timestamp is wrong for whatever reason, the file will be simply loaded again.

Doesn't it only reload the file if the file timestamp is newer than the saved timestamp? It's not unusual to see weird timestamps from the past or future with older FAT or networked filesystems or simply due to DTS: https://github.com/godotengine/godot/pull/74830 In practice it's not going to matter much, but I have found that using != for comparing file timestamps instead of < when trying to detect changed files will simply work correctly in more cases.

Reply:
tbh maybe the modified time thing should be removed altogether. Modifying the metadata externally while the editor is running is only useful when debugging code that uses metadata. If it's going to cause problems then it's not worth it.